threestudio/systems/zero123.py

```diff
@@ -112,48 +109,37 @@ def training_step(self, batch, batch_idx):
                     )  # [B, 2]
                     X = torch.linalg.lstsq(A, valid_pred_depth).solution  # [2, 1]
                     valid_gt_depth = A @ X  # [B, 1]
-                guidance_out["loss_depth"] = F.mse_loss(
+                set_loss("depth", F.mse_loss(
                     valid_gt_depth, valid_pred_depth
-                )
-        else:
-            guidance_out, guidance_eval_out = self.guidance(
-                out["comp_rgb"],
-                **batch,
-                rgb_as_latents=False,
-                guidance_eval=guidance_eval,
-            )
+                ))
+        elif guidance == "zero123":
+            # zero123
+            guidance_out, guidance_eval_out = self.guidance(out["comp_rgb"], **batch, rgb_as_latents=False, guidance_eval=guidance_eval)
+            # claforte: TODO: rename the loss_terms keys
+            set_loss("sds", guidance_out["loss_sds"])
```

Comment:
Good enough for now, but will need to add another `set_loss` line when using prolific_dreamer